### DISCUSSION TITLE
after cursor: empty value sometimes OK, sometimes not

### DISCUSSION THREAD

**Discussion by jfeltesse-mdsol** [community/community#24443] [Apps, API and Webhooks]
<p>Hi there,</p>
<p>A bit baffled by what I’m experiencing, hopefully I’m missing something.</p>
<p>In a nutshell, I have 2 queries, one to get tags on a repo and another one to perform a search.</p>
<p>Both of them implement pagination and as such I have to pass the afterCursor parameter.</p>
<p>On the first call since there’s no afterCursor I simply pass a blank string and it works for tags but not for search:</p>
<pre><code>query ($repoOwner: String!, $repoName: String!, $afterCursor: String!) {
  repository(owner: $repoOwner, name: $repoName) {
    name
    refs(refPrefix: "refs/tags/", first: 100, after: $afterCursor) {
      pageInfo {
        hasNextPage
        endCursor
      }
      // rest of the query here...
}

// variables:
{
  "repoOwner": "org name",
  "repoName": "repo name",
  "afterCursor": ""
}
</code></pre>
<p>yields expected results, while</p>
<pre><code>query ($searchQuery: String!, $afterCursor: String!) {
  search(type: REPOSITORY, query: $searchQuery, first: 10, after: $afterCursor) {
    pageInfo {
      hasNextPage
      endCursor
    }
    // rest of the query here...
}

// variables:
{
  "searchQuery": "some search",
  "afterCursor": ""
}
</code></pre>
<p>yields error</p>
<p>“type”: “INVALID_CURSOR_ARGUMENTS”,</p>
<p>“message”: “`` does not appear to be a valid cursor.”</p>
<p>Do acceptable values for afterCursor depend on the connection?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Have you tried <code>null</code> instead of a blank string? It’s possible that the implementation slightly varies, yes. You may also want to try using two copies of the query, one with the <code>after</code> parameter and one without.</p>

**Comment by jfeltesse-mdsol** [comment]
<p>Making $afterCursor a String (not String!) and passing a null works!</p>
<p>Thanks!!</p>
